src/main/java/org/osc/controller/nsc/utils/RedirectionApiUtils.java

```diff
@@ -168,18 +168,45 @@ public NetworkElementEntity txNetworkElementEntityById(Long id) {
         return this.em.find(NetworkElementEntity.class, id);
     }
 
-    public void removeSingleInspectionHook(InspectionHookEntity inspectionHookEntity) {
+    public void removeSingleInspectionHook(String hookId) {
+        InspectionHookEntity inspectionHook = this.txControl.required(() -> {
+            return this.em.find(InspectionHookEntity.class, hookId);
+        });
+
+        NetworkElementEntity inspected = inspectionHook.getInspectedPort();
+
         this.txControl.required(() -> {
-            NetworkElementEntity networkElementEntity = inspectionHookEntity.getInspectedPort();
+            InspectionHookEntity dbInspectionHook  =
+            this.em.find(InspectionHookEntity.class, hookId);
+            NetworkElementEntity dbInspectedPort = dbInspectionHook.getInspectedPort();
 
-            inspectionHookEntity.setInspectionPort(null);
-            inspectionHookEntity.setInspectedPort(null);
-            networkElementEntity.setInspectionHook(null);
-            this.em.remove(inspectionHookEntity);
+            dbInspectedPort.setInspectionHook(null);
+            dbInspectionHook.setInspectedPort(null);
+            return null;
+        });
+
+        this.txControl.required(() -> {
+            NetworkElementEntity tmpNE = this.em.find(NetworkElementEntity.class, inspected.getElementId());
+
+            this.em.remove(tmpNE);
+
+            Query q = this.em.createQuery("DELETE FROM InspectionHookEntity WHERE hook_id = :id");
+            q.setParameter("id", hookId);
+            q.executeUpdate();
             return null;
         });
     }
 
+    public void removeSingleInspectionPort(String inspectionPortId) {
+        this.txControl.required(() -> {
```

Comment:
why do we need to use query here? why cant we load from em?

Reply:
@dmitryintel can you provide rationale between when to choose query vs using the em?

We need to consistently use the rationale to determine one or the other. I see a lot of places where query can be replaced with em...